src/Lexer.js

```diff
@@ -67,33 +62,30 @@ const commentRegex = new RegExp(`^${commentRegexString}`);
 /** Main Lexer class */
 export default class Lexer implements LexerInterface {
     input: string;
-    pos: number;
+    tokenRegex: RegExp;
 
     constructor(input: string) {
         // Separate accents from characters
         this.input = input;
-        this.pos = 0;
+        this.tokenRegex = new RegExp(tokenRegexString, 'g');
```

Comment:
It's cool that this works even when `feed`ing another string, thanks to the semantics of `exec`.

Reply:
Oops, I was wrong about this.  I just tried the following in Chrome:

```js
s='aaaaaaa'
re.exec(s)  // ["a", index: 0, input: "aaaaaaa", groups: undefined]
re.exec(s)  // ["a", index: 1, input: "aaaaaaa", groups: undefined]
re.exec(s)  // ["a", index: 2, input: "aaaaaaa", groups: undefined]
t='aaaa'
re.exec(t)  // ["a", index: 3, input: "aaaa", groups: undefined]
```

It seems we should set `this.tokenRegex.lastIndex = 0` in `feed`.

Reply:
@edemaine Is it possible to feed new string to the lexer? Currently, new `RegExp` instance is created when instantiating lexer.

Reply:
I don't believe so.

Reply:
Oops, indeed, all is well. I thought `Lexer` had the `feed` method but it's `MacroExpander`, which creates a new `Lexer` each time.